### PR TITLE
TCP protocol support.

### PR DESCRIPTION
**Pull request by firerain-fd**
TCP protocol support has been added.
Updated tests.

https://github.com/justeattakeaway/JustEat.StatsD/issues/537

### CODE REVIEW

File: tests/JustEat.StatsD.Tests/Utf8TagsFormatterTests.cs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using System.Linq.Expressions;
```
Comment (martincostello) [inline tests/JustEat.StatsD.Tests/Utf8TagsFormatterTests.cs:1]: What's this used for?

**Review by martincostello — COMMENTED** [review]
Thanks for your contribution. It looks like some of the tests are failing though.

File: tests/JustEat.StatsD.Tests/Utf8TagsFormatterTests.cs (hunk 1 above)
Comment (firerain-fd) [reply tests/JustEat.StatsD.Tests/Utf8TagsFormatterTests.cs]: Sorry, Visual Studio autocomplete.

**Review by martincostello — APPROVED** [review]

**Comment by firerain-fd** [comment]
Yes, I checked this in my project. Now I can get an exception when it is impossible to connect to statsd server or when I fail to send a request.

**Pull request merged**